File: server/src/Reducers/Items.cs
using pillz.server.Tables;
using SpacetimeDB;

namespace pillz.server.Reducers;

public partial class Items
{
    [Reducer]
    public static void UpdateAmmo(ReducerContext ctx, uint id, DbVector2 position)
    {
        var ammo = ctx.Db.Entity.Id.Find(id);
        if (ammo != null)
        {
            ammo = ammo.Value with { Position = position };
            ctx.Db.Entity.Id.Update(ammo.Value);
        }
    }
    
    [Reducer]
    public static void DeleteAmmo(ReducerContext ctx, uint id)
    {
        var ammo = ctx.Db.Ammo.EntityId.Find(id) ?? throw new Exception("Ammo not found");

        ctx.Db.Entity.Id.Delete(id);
        ctx.Db.Ammo.EntityId.Delete(ammo.EntityId);
    }
}

File: server/src/reducers/items.rs
use crate::dto::item_movement::ItemMovement;
use crate::tables::ammo::ammo;
use crate::tables::entity::entity;
use spacetimedb::ReducerContext;

#[spacetimedb::reducer]
pub fn update_ammo(ctx: &ReducerContext, id: u32, movement: ItemMovement) {
    if let Some(mut ammo) = ctx.db.ammo().entity_id().find(id) {
        ammo.position = movement.position;
        ammo.direction = movement.direction;
        ctx.db.ammo().entity_id().update(ammo);

        if let Some(mut e) = ctx.db.entity().id().find(id) {
            e.position = movement.position;
            ctx.db.entity().id().update(e);
        }

        // debug!(
        //     "({})Updated ammo type ({:?}) at ({}, {}) with id: {}.",
        //     player.id, clone.ammo_type, clone.position.x, clone.position.y, clone.entity_id
        // );
    }
}

#[spacetimedb::reducer]
pub fn delete_ammo(ctx: &ReducerContext, id: u32) {
    let ammo = ctx.db.ammo().entity_id().find(id).expect("Ammo not found");

    ctx.db.entity().id().delete(id);
    ctx.db.ammo().entity_id().delete(ammo.entity_id);
}
